src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -129,24 +130,52 @@ private static int KMPSearch(byte[] pattern, byte[] bytes)
                     }
                 }
             }
+
             return -1;
         }
 
-        private static void SearchAndReplace(byte[] array, byte[] searchPattern, byte[] patternToReplace, string appHostSourcePath)
+        private static unsafe void SearchAndReplace(
+            MemoryMappedViewAccessor accessor,
+            byte[] searchPattern,
+            byte[] patternToReplace,
+            string appHostSourcePath)
         {
-            int offset = KMPSearch(searchPattern, array);
-            if (offset < 0)
+            byte* bytes = null;
+            accessor.SafeMemoryMappedViewHandle.AcquirePointer(ref bytes);
+            bytes = bytes + accessor.PointerOffset;
+
+            try
             {
-                throw new BuildErrorException(Strings.AppHostHasBeenModified, appHostSourcePath, _placeHolder);
-            }
+                int position = KMPSearch(searchPattern, bytes, accessor.Capacity);
+                if (position < 0)
+                {
+                    throw new BuildErrorException(Strings.AppHostHasBeenModified, appHostSourcePath, _placeHolder);
+                }
 
-            patternToReplace.CopyTo(array, offset);
+                accessor.WriteArray(
+                    position: position,
+                    array: patternToReplace,
+                    offset: 0,
+                    count: patternToReplace.Length);
 
+                Pad0(searchPattern, patternToReplace, bytes, position);
+            }
+            finally
+            {
+                if (accessor.SafeMemoryMappedViewHandle != null)
```

Comment:
Sorry. missed this before, but the thing that you are expected to check for null here is the pointer. And AcquirePointer is supposed to be in the try. 

See how it is documented to use AcquirePointer here: https://source.dot.net/#System.Private.CoreLib/shared/System/Runtime/InteropServices/SafeBuffer.cs,17340e3be744c19e,references

So in this case, it would be:
``` C#

byte* pointer = null;
try
{
     accessor.SafeMemoryMappedViewHandle.AcquirePointer(ref pointer);
     byte *bytes = pointer + accessor.PointerOffset;
     // use bytes
}
finally
{
    if (pointer != null)
    {
        accessor.SafeMemoryMappedViewHandle.ReleasePointer();
    }
}
```

Reply:
Sorry, this should be good now. Edit according to your code